plotutils/graph.h

```diff
@@ -0,0 +1,205 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2014 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+// TODO(dkorolev): Platform-specific implementation and guards.
+
+#ifndef BRICKS_PLOTUTILS_GRAPH_H
+#define BRICKS_PLOTUTILS_GRAPH_H
+
+#include <cassert>
+#include <vector>
+#include <string>
+#include <map>
+
+#include "../file/file.h"
+#include "../strings/util.h"
+
+namespace bricks {
+namespace plotutils {
+
+// https://www.gnu.org/software/plotutils/manual/en/html_node/Plot-Options.html#Plot-Options
+// 0: no axes, tick marks or labels.
+// 1: a pair of axes, with tick marks and labels.
+// 2: box around plot, with tick marks and labels.
+// 3: box around plot, with tick marks and labels; also grid lines.
+// 4: axes intersect at the origin, with tick marks and labels.
+enum class GridStyle : int { None = 0, Axes = 1, Box = 2, Full = 3, AxesAtOrigin = 4 };
+
+// https://www.gnu.org/software/plotutils/manual/en/html_node/Dataset-Options.html#Dataset-Options
+// By convention, linemode #0 means no line at all (data points are disconnected).
+// 1: red, solid
+// 2: green, solid
+// 3: blue, solid
+// 4: magenta, solid
+// 5: cyan, solid
+// Linemodes #6 through #10 use the same five colors, but are dotted; linemodes #11 through #15 are dotdashed;
+// linemodes #16 through #20 are shortdashed; and linemodes #21 through #25 are longdashed.
+// So besides linemode #0, there are a total of 25 distinct colored linemodes
+enum class LineMode : int {
+  None = 0,
+};
+enum class LineColor : int { Red = 1, Green = 2, Blue = 3, Magenta = 4, Cyan = 5 };
```

Comment:
I did not realized first this is for a third-party command line tool. It's OK then.